File: backend/schemas/application.py
from enum import Enum
from typing import Optional

from pydantic import BaseModel, Field

from .base import PaginatedResponse, TimestampedModel


class ApplicationStatus(str, Enum):
    PENDING = "pending"
    INTERVIEW = "interview"
    HIRED = "hired"
    REJECTED = "rejected"
    WITHDRAWN = "withdrawn"


class JobSummary(BaseModel):
    title: str
    company_name: str
    company_id: str


class ApplicationBase(TimestampedModel):
    job_id: str
    worker_id: str
    cover_letter: Optional[str] = Field(default=None, max_length=4000)
    status: ApplicationStatus = ApplicationStatus.PENDING


class ApplicationCreate(BaseModel):
    job_id: str
    cover_letter: Optional[str] = Field(default=None, max_length=4000)


class ApplicationUpdate(BaseModel):
    status: Optional[ApplicationStatus] = None
    cover_letter: Optional[str] = Field(default=None, max_length=4000)


class ApplicationRead(ApplicationBase):
    id: str
    job: Optional[JobSummary] = None


class ApplicationList(PaginatedResponse[ApplicationRead]):
    pass

File: frontend/lib/core/models/application.dart
class Application {
  const Application({
    required this.id,
    required this.jobId,
    required this.workerId,
    required this.status,
    this.coverLetter,
  });

  factory Application.fromJson(Map<String, dynamic> json) => Application(
        id: json['id'] as String,
        jobId: json['job_id'] as String? ?? json['jobId'] as String? ?? '',
        workerId: json['worker_id'] as String? ?? json['workerId'] as String? ?? '',
        status: json['status'] as String,
        coverLetter: json['cover_letter'] as String?,
      );

  final String id;
  final String jobId;
  final String workerId;
  final String status;
  final String? coverLetter;
}
